framework/quota/src/org/apache/cloudstack/quota/vo/ServiceOfferingVO.java

```diff
@@ -310,14 +269,12 @@ public String getDeploymentPlanner() {
     }
 
     public String getDetail(String name) {
-        assert (details != null) : "Did you forget to load the details?";
-
-        return details != null ? details.get(name) : null;
+        if (details == null) throw new IllegalStateException("Cannot set value as details is null");
+        return details.get(name);
     }
 
     public void setDetail(String name, String value) {
-        assert (details != null) : "Did you forget to load the details?";
-
+        if (details == null) throw new IllegalStateException("Cannot set value as details is null");
```

Comment:
We can check that, but if we will let it as it is. 
 I think it should have a test case written to check that behavior.